include/signal.h

```diff
@@ -0,0 +1,50 @@
+#ifndef _SYS_SIGNAL_H_
+#define _SYS_SIGNAL_H_
+
+#include <common.h>
+#include <bitstring.h>
+#include <queue.h>
+#include <mutex.h>
+#include <signum.h>
+
+typedef struct proc proc_t;
+
+typedef bitstr_t sigset_t[bitstr_size(SIG_LAST)];
+
+typedef void (*sighandler_t)(int);
+
+#define SIG_DFL (sighandler_t *)0x00
+#define SIG_IGN (sighandler_t *)0x01
+#define SIG_TERM (sighandler_t *)0x02
```

Comment:
Ok. I see the point. Your observation tackles the difference between Linux and BSD worlds. *Linux* is a kernel, whereas **BSD* is an operating system. In the latter case user-space and kernel-space are tightly coupled and delivered as single product. In the former you have a choice of wildly different user-spaces (Android, ChromeOS, Debian, etc.) running on top of same kernel.

*In the long run I want Mimiker to follow BSD philosophy and be a complete OS.*